vsintegration/src/FSharp.Editor/CodeFixes/MakeOuterBindingRecursive.fs

```diff
@@ -9,45 +9,38 @@ open System.Collections.Immutable
 open Microsoft.CodeAnalysis.Text
 open Microsoft.CodeAnalysis.CodeFixes
 
+open CancellableTasks
+
 [<ExportCodeFixProvider(FSharpConstants.FSharpLanguageName, Name = CodeFix.MakeOuterBindingRecursive); Shared>]
 type internal MakeOuterBindingRecursiveCodeFixProvider [<ImportingConstructor>] () =
     inherit CodeFixProvider()
 
     override _.FixableDiagnosticIds = ImmutableArray.Create("FS0039")
 
-    override _.RegisterCodeFixesAsync context =
-        asyncMaybe {
-            let! parseResults =
-                context.Document.GetFSharpParseResultsAsync(nameof (MakeOuterBindingRecursiveCodeFixProvider))
-                |> liftAsync
-
-            let! sourceText = context.Document.GetTextAsync(context.CancellationToken)
-
-            let diagnosticRange =
-                RoslynHelpers.TextSpanToFSharpRange(context.Document.FilePath, context.Span, sourceText)
-
-            do! Option.guard (parseResults.IsPosContainedInApplication diagnosticRange.Start)
-
-            let! outerBindingRange = parseResults.TryRangeOfNameOfNearestOuterBindingContainingPos diagnosticRange.Start
-            let! outerBindingNameSpan = RoslynHelpers.TryFSharpRangeToTextSpan(sourceText, outerBindingRange)
-
-            // One last check to verify the names are the same
-            do!
-                Option.guard (
-                    sourceText
-                        .GetSubText(outerBindingNameSpan)
-                        .ContentEquals(sourceText.GetSubText(context.Span))
-                )
-
-            let title =
-                String.Format(SR.MakeOuterBindingRecursive(), sourceText.GetSubText(outerBindingNameSpan).ToString())
-
-            do
-                context.RegisterFsharpFix(
-                    CodeFix.MakeOuterBindingRecursive,
-                    title,
-                    [| TextChange(TextSpan(outerBindingNameSpan.Start, 0), "rec ") |]
-                )
-        }
-        |> Async.Ignore
-        |> RoslynHelpers.StartAsyncUnitAsTask(context.CancellationToken)
+    override this.RegisterCodeFixesAsync context = context.RegisterFsharpFix(this)
+
+    interface IFSharpCodeFixProvider with
+        member _.GetCodeFixIfAppliesAsync context =
+            cancellableTask {
+                let! parseResults = context.GetParseResultsAsync(nameof MakeOuterBindingRecursiveCodeFixProvider)
+                let! sourceText = context.GetSourceTextAsync()
+                let! diagnosticRange = context.GetErrorRangeAsync()
+
+                return
+                    Option.guard (parseResults.IsPosContainedInApplication diagnosticRange.Start)
```

Comment:
`Option.guard` did make some sense within the asyncMaybe CE - it was an inner compact DSL.
Without it, it is just a glorified if.

=> let's use a regular explicit if, like anywhere else (where we are not in a asyncMaybe context)

Reply:
IMO this is also a personal preference - I love pipes and hate ifs, and the former are more readable for me. I'll change it here though since `Option.guard` is not part of the standard library.